frame/democracy/src/lib.rs

```diff
@@ -477,6 +484,29 @@ pub mod pallet {
 		Seconded { seconder: T::AccountId, prop_index: PropIndex },
 		/// A proposal got canceled.
 		ProposalCanceled { prop_index: PropIndex },
+		/// Metadata for a proposal or a referendum has been set.
+		MetadataSet {
+			/// Metadata owner.
+			owner: MetadataOwner,
+			/// Preimage hash.
+			hash: PreimageHash,
+		},
+		/// Metadata for a proposal or a referendum has been cleared.
+		MetadataCleared {
+			/// Metadata owner.
+			owner: MetadataOwner,
+			/// Preimage hash.
+			hash: PreimageHash,
+		},
+		/// Metadata has been reset to new owner.
+		MetadataReset {
```

Comment:
Reset sounds to me like clearing the Metadata, as opposed to transferring it to a new owner. WDYT?

Reply:
Yeah, probably just because it's used more frequently as 'clear', but my meaning 'reset' - is set again, anew. Plus the arguments make it more clear.
I am happy to rename if you have better option?

Reply:
I'm not particular about it. It's just not clear what it does until someone reads the doc. Would something like `MetadataTransfered` not work?

Reply:
In blockchain context transferred has its own flavor (:
When you see the event, without reading the doc, you arguments, prev_owner and owner

Reply:
What do you think about the `MetadataOwnerChanged`? `MetadataReset` sounds very ambiguous here